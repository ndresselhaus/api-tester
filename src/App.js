import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import {parse as qsParse} from 'qs';

import UrlInput from './UrlInput';
import RequestBodyInput from './RequestBodyInput';
import HttpMethodInput from './HttpMethodInput';

class App extends Component {
    constructor() {
        super();
        this.state = {
            url: '',
            requestBodyInput: '',
        };
    }

    handleChange = (key, e) => {
        this.setState({[key]: e.target.value});
    };

    generateThings = () => {
        const {url, requestBodyInput: requestBody} = this.state;
        const queryParams = qsParse(url.substring(url.indexOf('?') + 1));
        const bodyParams = JSON.parse(requestBody);

        this.setState({queryParams, bodyParams});

    };

    render() {
        return (
            <div className="App">
                <div className="App-header">
                    <img src={logo} className="App-logo" alt="logo"/>
                    <h2>Welcome to React</h2>
                </div>
                <p className="App-intro">
                    <UrlInput url={this.state.url} onChange={(e) => this.handleChange('url', e)}/>
                </p>
                <div>
                    <RequestBodyInput
                        requestBody={this.state.requestBody}
                        onChange={(e) => this.handleChange('requestBodyInput', e)}/>
                </div>
                <div>
                    <button onClick={this.generateThings}>Generate All The Things</button>
                </div>
                <div style={{padding: 5, border: 1}}>
                    {this.state.queryParams && Object.keys(this.state.queryParams).map(parameter => {
                        return <label key={parameter}>{parameter}<input type="text"/></label>
                    })}
                    {this.state.bodyParams && Object.keys(this.state.bodyParams).map(bodyParam => {
                            return <label key={bodyParam}>{bodyParam}<input type="text"/></label>
                        }
                    )}
                </div>
                <div>
                    <HttpMethodInput httpMethod={this.state.httpMethod}
                                     onChange={e => this.handleChange('httpMethod', e)}/>
                </div>
            </div>
        );
    }
}

export default App;
