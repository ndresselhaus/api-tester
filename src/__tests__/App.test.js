import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import RequestBodyInput from '../RequestBodyInput';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('renders snapshots with shallow enzyme', () => {
    expect(shallow(<App />)).toMatchSnapshot();
});

it('renders with react-test-renderer', () => {
    expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
});

it('updates request body state', () => {
    const app = shallow(<App />);
    const rbi = app.find(RequestBodyInput);
    rbi.props().onChange({target: {value: 'foo'}});
    expect(app.state().requestBodyInput).toEqual('foo');
});