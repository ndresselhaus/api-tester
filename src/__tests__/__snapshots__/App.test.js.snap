// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshots with shallow enzyme 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    <UrlInput
      onChange={[Function]}
      url=""
    />
  </p>
  <div>
    <RequestBodyInput
      onChange={[Function]}
    />
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Generate Things
    </button>
  </div>
  <div>
    <HttpMethodInput
      httpMethod="GET"
      onChange={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "border": 1,
        "padding": 5,
      }
    }
  >
    <ParameterList
      onChange={[Function]}
      title="Query Parameters"
    />
    <ParameterList
      onChange={[Function]}
      title="Body Parameters"
    />
  </div>
</div>
`;

exports[`renders with react-test-renderer 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    <label>
      URL
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </p>
  <div>
    <textarea
      className="App-request-body-input"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Generate Things
    </button>
  </div>
  <div>
    <label>
      Http Method
      <select
        onChange={[Function]}
        value="GET"
      >
        <option
          value="GET"
        >
          GET
        </option>
        <option
          value="POST"
        >
          POST
        </option>
      </select>
    </label>
  </div>
  <div
    style={
      Object {
        "border": 1,
        "padding": 5,
      }
    }
  />
</div>
`;
